internal/featuretests/v3/tlsroute_test.go

```diff
@@ -0,0 +1,170 @@
+// Copyright Project Contour Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package v3
+
+import (
+	"testing"
+
+	envoy_listener_v3 "github.com/envoyproxy/go-control-plane/envoy/config/listener/v3"
+	envoy_discovery_v3 "github.com/envoyproxy/go-control-plane/envoy/service/discovery/v3"
+	"github.com/projectcontour/contour/internal/dag"
+	envoy_v3 "github.com/projectcontour/contour/internal/envoy/v3"
+	"github.com/projectcontour/contour/internal/fixture"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/intstr"
+	"k8s.io/utils/pointer"
+	gatewayapi_v1alpha1 "sigs.k8s.io/gateway-api/apis/v1alpha1"
+)
+
+func TestTLSRoute(t *testing.T) {
+	rh, c, done := setup(t)
+	defer done()
+
+	//s1 := &v1.Secret{
```

Comment:
the docs mention that a secret is required on TLSRoutes, seems like the integration test and this one have omitted it and not made the TLS secret required?

Reply:
is this because this just enables passthrough TLS and not terminating and re-encrypting?

Reply:
what are the rules that we/upstream decided on for supporting TLS passthrough/termination and how they relate to TLSRoute/TCPRoute?

Reply:
There were some discussions upstream (https://github.com/kubernetes-sigs/gateway-api/pull/633#discussion_r622203260) and essentially from what I understand, `TLSRoute` is a passthrough TCPProxy against SNI. 

If you want to terminate then route, you'd have to use TCPRoute. 

@youngnick said we can't support TCPRoute so this will be a point that differs from HTTPProxy. 

- HTTPProxy.TCPProxy (passthrough: true) == TLSRoute
- HTTPProxy.TCPProxy (passthrough: false) == TCPRoute